### PR TITLE
Remove all assumptions about closure response keys being present

### PR DESCRIPTION
**Pull request by BugsNash**
Removes assumptions in build script about what keys are present in closure response object to prevent crashes when these assumptions are violated. E.g. a crash like this was caused in #374 where material design library warnings were missing the ‘line’ key. Now when keys are missing that information in the build output is replaced by a warning that the key is missing.

[Diff](https://gist.github.com/BugsNash/337cfc2f9a74aeef9bb43c0c7f8a169c/revisions) of build output of this script on #374

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -157,17 +157,24 @@ def process_index(out_dir):
...
       text = error['warning']
-    print(filename + ':' + str(error['lineno']) + ' ' + text)
-    print(error['line'])
+    else:
+      text = missing_key_string('error/warning')
+    print(filename + ':' + str(error.get('lineno', missing_key_string('lineno'))) + ' ' + text)
+    print (error.get('line', missing_key_string('line')))
+
+
+def missing_key_string(key):
```
Comment (E-T-) [inline build.py]: This function's name makes it hard to read. I see that

`missing_key_string('foo')`

will return

`['foo' key missing]`

but it seems like a weird name. Maybe something like `get_missing_key_msg` makes more sense?

File: build.py [hunk 2]
```diff
@@ -225,7 +232,10 @@ def compile_js(out_path, js_files, level, externs):
 
   print('Writing', out_path)
   os.makedirs(os.path.dirname(out_path), exist_ok=True)
-  open(out_path, 'w').write(result['compiledCode'])
+  if result.get('compiledCode'):
+    open(out_path, 'w').write(result.get('compiledCode'))
+  else:
+    print(missing_key_string('compiledCode'))
```
Comment (E-T-) [inline build.py]: How odd! In what case do you miss this key in the response?

**Comment by BugsNash** [comment]
Oh yeah sure I can add a diff of the build output. I didn't do it cause on master there is no difference, but if i run the script on #374 you can see the difference. The output will be massive so maybe I will just include a snippet or something.

**Comment by BugsNash** [comment]
So I have actually encountered build script crashes becuase 'compiledCode' key was missing a couple of times, I don't know why it was missing and I can't reliably replicate it :\

**Comment by BugsNash** [comment]
I have added a link to the build diff in the description.

File: build.py (hunk 2 above)
Comment (BugsNash) [reply build.py]: This case would actually make more sense to pring a more descriptive error message saying that compilation failed. I might write another patch to check for server errors in the closure response, as they might expain the reason why this is sometimes missing?

File: build.py (hunk 1 above)
Comment (BugsNash) [reply build.py]: Done

**Pull request merged**